Spigot-Server-Patches/0336-Extend-player-profile-API-to-support-skin-changes.patch

```diff
@@ -0,0 +1,191 @@
++    public int getDimension() {
++        return dimension;
++    }
++    // Paper end
++
+     // CraftBukkit start
+     @Override
+     public TileEntity getTileEntity(BlockPosition pos) {
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+index 210e3bc4..b0335684 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftWorld.java
+@@ -101,6 +101,12 @@ public class CraftWorld implements World {
+     }
+     // Paper end
+ 
++    // Paper start - Provide dimension number of world
++    public int getDimension() {
++        return world.dimension;
++    }
++    // Paper end
++
+     private static final Random rand = new Random();
+ 
+     public CraftWorld(WorldServer world, ChunkGenerator gen, Environment env) {
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index 6cbf429f..bd743be5 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -55,6 +55,7 @@ import org.bukkit.craftbukkit.CraftStatistic;
+ import org.bukkit.craftbukkit.CraftWorld;
+ import org.bukkit.craftbukkit.advancement.CraftAdvancement;
+ import org.bukkit.craftbukkit.advancement.CraftAdvancementProgress;
++import org.bukkit.craftbukkit.inventory.CraftItemStack;
+ import org.bukkit.craftbukkit.map.CraftMapView;
+ import org.bukkit.craftbukkit.map.RenderData;
+ import org.bukkit.craftbukkit.scoreboard.CraftScoreboard;
+@@ -1172,14 +1173,102 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+     public void setPlayerProfile(PlayerProfile profile) {
+         EntityPlayer self = getHandle();
+         self.setProfile(CraftPlayerProfile.asAuthlibCopy(profile));
+-        List<EntityPlayer> players = server.getServer().getPlayerList().players;
+-        for (EntityPlayer player : players) {
+-            player.getBukkitEntity().reregisterPlayer(self);
+-        }
++        refreshPlayer();
+     }
+     public PlayerProfile getPlayerProfile() {
+         return new CraftPlayerProfile(this).clone();
+     }
++
++    private void refreshPlayer() {
++        //Calculate the correct dimension
++        int dimension = (int) ((WorldServer)getHandle().getWorld()).getDimension();
++        if (dimension == 8 || dimension == 13)
++            dimension = 0;
++        if (dimension == 1 || dimension == 14)
++            dimension = -1;
++        if (dimension == 16 || dimension == 12)
++            dimension = 1;
++
++        //Respawn packet
++        Packet respawn = new PacketPlayOutRespawn(dimension, ((WorldServer)getHandle().getWorld()).worldData.getDifficulty(), ((WorldServer)getHandle().getWorld()).worldData.getType(), EnumGamemode.getById(getHandle().playerInteractManager.getGameMode().getId()));
++
++        Location l = getLocation();
++
```

Comment:
All of this should be unnecessary. We retrack the entity already to other players, which picks up the changes. The scope of this PR is trying to solve the player seeing their own change.

Why would the player need help seeing new equipment? It's just the profile we are trying here.

Shouldn't the respawn packet be the main need.